python/rapidsmpf/rapidsmpf/streaming/cudf/table_chunk.pyx

```diff
@@ -330,3 +382,69 @@ cdef class TableChunk:
         True if the table chunk can be spilled, otherwise, False.
         """
         return deref(self.handle_ptr()).is_spillable()
+
+
+@singledispatch
```

Comment:
*suggestion*: I think there's no need for `get_table_chunk` to accept a `Message`, since the thing it does in that case (`chunk = TableChunk.from_message(msg)`) is trivially available to the user.

Then this utility becomes a method of `TableChunk` that wraps up `make_available` with obtaining a reservation.

Could we instead implement a `to_device` utility method on `TableChunk`?

```
def to_device(self, BufferResource br not None, *, allow_overbooking):
    cdef MemoryReservation res = br.reserve_and_spill(...)
    return self.make_available(res)
```

Reply:
@wence- WDYT?